search_gov_crawler/search_gov_spiders/pipelines.py

```diff
@@ -2,46 +2,99 @@
-        self.base_path_name = str(self.parent_file_path / f"output/all-links-p{os.getpid()}.csv")
-        self.short_file = open(self.base_path_name, "a", encoding="utf-8")
-        self.max_file_size = 39000000 #3.9MB max
-        self.paginate = True
+        self.file_path = None
+        self.current_file = None
+
+        if not self.api_url:
+            output_dir = Path(__file__).parent.parent / "output"
+            output_dir.mkdir(parents=True, exist_ok=True)
+            base_filename = f"all-links-p{self.APP_PID}"
+            self.file_path = output_dir / f"{base_filename}.csv"
+            self.current_file = open(self.file_path, "a", encoding="utf-8")
+
+    def process_item(self, item, spider):
+        """Handle each item by writing to file or batching URLs for an API POST."""
+        url = item.get("url", "")
+        if not url:
+            raise DropItem("Missing URL in item")
+
+        if self.api_url:
+            self._process_api_item(url, spider)
+        else:
+            self._process_file_item(url)
 
-    def process_item(self, item, _spider):
-        """Checks that the file is not at max size.
-        Adds it to the file if less, or creates a new file if too large."""
-        line = item["url"]
-        self.current_file_size += 1
-        file_stats = os.stat(self.base_path_name)
-        self.current_file_size += file_stats.st_size
-        next_file_size = self.current_file_size + len(line)
-        if self.paginate and next_file_size > self.max_file_size:
-            self.short_file.close()
-            new_name = str(self.parent_file_path / f"output/all-links-p{os.getpid()}-{self.file_number}.csv")
-            os.rename(self.base_path_name, new_name)
-            self.file_number = self.file_number + 1
-            self.short_file = open(self.base_path_name, "w", encoding="utf-8")
-            self.current_file_size = 0
-        self.short_file.write(line)
-        self.short_file.write("\n")
-        self.current_file_size = self.current_file_size + len(line)
         return item
 
+    def _process_api_item(self, url, spider):
+        """Batch URLs for API and send POST if size limit is reached."""
+        self.urls_batch.append(url)
+        if self._batch_size() >= self.MAX_FILE_SIZE_BYTES:
+            self._send_post_request(spider)
+
+    def _process_file_item(self, url):
+        """Write URL to file and rotate the file if size exceeds the limit."""
+        self.current_file.write(f"{url}\n")
+        if self._file_size() >= self.MAX_FILE_SIZE_BYTES:
+            self._rotate_file()
+
+    def _batch_size(self):
+        """Calculate total size of the batched URLs."""
+        return sum(len(url.encode("utf-8")) for url in self.urls_batch)
+
+    def _file_size(self):
+        """Get the current file size."""
+        self.current_file.flush()  # Ensure the OS writes buffered data to disk
+        return self.file_path.stat().st_size
+
+    def _rotate_file(self):
```

Comment:
We need a way to identify the process /spider calling the file. This was done with my changes from https://github.com/GSA/searchgov-spider/pull/60 I recommend putting back my changes